lib/byte-to-hex-table/default.nix

```diff
@@ -0,0 +1,56 @@
+# Generates a lookup table mapping bytes (as single-byte strings) to their
+# hexadecimal representation (as two-character uppercase strings).
+#
+# This is used by lib.strings.escapePythonBytes to convert arbitrary byte
+# sequences into Python bytes literals.
+
+let
+  # Hex digits
+  hexDigits = [
+    "0"
+    "1"
+    "2"
+    "3"
+    "4"
+    "5"
+    "6"
+    "7"
+    "8"
+    "9"
+    "A"
+    "B"
+    "C"
+    "D"
+    "E"
+    "F"
+  ];
+
+  # Generate a two-digit hex string from a number (0-255).
+  toHex =
+    n:
+    let
+      high = n / 16;
+      low = n - (high * 16);
+    in
+    builtins.elemAt hexDigits high + builtins.elemAt hexDigits low;
```

Comment:
Could we use `lib.toHexString` and `lib.fromHexString` here?

Reply:
I just pushed a new version of this pull request that uses `lib.trivial.toHexString`. The new version of this pull request also renames the `toHex` function to `toTwoDigitHexString` in order to highlight the difference between that function and `lib.trivial.toHexString`. That being said, I wasn’t able to find a place to use `lib.trivial.fromHexString`. What did you have in mind for `lib.trivial.fromHexString`?